jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java

```diff
@@ -83,6 +84,15 @@
     private UpgradeResponse upgradeResponse;
     private CompletableFuture<Session> openFuture;
 
+    static
+    {
+        // Attempt to discover a RemoteEndpointFactory with the SerivceLoader.
+        Iterator<RemoteEndpointFactory> iter = ServiceLoader.load(RemoteEndpointFactory.class).iterator();
```

Comment:
Is this running under the correct Classloader / Thread Context?

Reply:
Yes, its running on the same thread and with the same contextClassLoader it has for `doStart()`.